manager/allocator/networkallocator/networkallocator.go

```diff
@@ -186,12 +187,28 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 		return
 	}
 
-	// First allocate VIPs for all the pre-populated endpoint attachments
-	for _, eAttach := range s.Endpoint.VirtualIPs {
-		if err = na.allocateVIP(eAttach); err != nil {
-			return
+	// Allocate VIPs for all the pre-populated endpoint attachments
+	eVIPs := s.Endpoint.VirtualIPs[:0]
```

Comment:
@aboch not really. The advantage is mentioned here https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating
I was just following that convention